Mixpanel (Export Events) Analytics (Source) initial commit

mohan-h opened the pull request:
Mixpanel (Export Events) Analytics (Source) initial commit

carlojavier commented on help/sources/tutorials/ui/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,121 @@
+
+[Mixpanel](https://www.mixpanel.com) is a product analytics tool that enables capturing data on how users interact with a digital product. Mixpanel allows you to analyze this product data with simple, interactive reports that let you query and visualize the data with just a few clicks.
+
+This Adobe Experience Platform [sources](https://experienceleague.adobe.com/docs/experience-platform/sources/home.html?lang=en) leverages the [Event Export API > Download](https://developer.mixpanel.com/reference/raw-event-export) to download your event data as it is received and stored within Mixpanel, along with all event properties (including distinct_id) and the exact timestamp the event was  ...
```
- Let's rephrase the first part to: "The Mixpanel [source](link) leverages..."
- Let's put `distinct_id` in backticks in the last sentence

mohan-h replied:
The 1st bullet would read odd, have updated _[Event Export API > Download]_ to _[Mixpanel Event Export API > Download]_. The Mixpanel prefix was missed.
2nd bullet, done.

carlojavier commented on help/sources/tutorials/ui/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,121 @@
+
+[Mixpanel](https://www.mixpanel.com) is a product analytics tool that enables capturing data on how users interact with a digital product. Mixpanel allows you to analyze this product data with simple, interactive reports that let you query and visualize the data with just a few clicks.
+
+This Adobe Experience Platform [sources](https://experienceleague.adobe.com/docs/experience-platform/sources/home.html?lang=en) leverages the [Event Export API > Download](https://developer.mixpanel.com/reference/raw-event-export) to download your event data as it is received and stored within Mixpanel, along with all event properties (including distinct_id) and the exact timestamp the event was  ...
```
Let's rephrase the first part to: "The Mixpanel [source](link) leverages..."

mohan-h replied:
The 1st bullet would read odd, have updated [Event Export API > Download] to [Mixpanel Event Export API > Download]. The Mixpanel prefix was missed.

carlojavier commented on help/sources/tutorials/ui/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,121 @@
+
+## Prerequisites
+
+Before you start configuring the extension you need to have a Mixpanel account. If you do not have one already, go to the Mixpanel [register](https://mixpanel.com/register/) page to register and create your Mixpanel account.
```
Let's rephrase this to: "The first step in creating a Mixpanel source connection is to ensure that you have a Mixpanel account. If you do not have one already, see the [Mixpanel register](link) page to register and create your account."

carlojavier replied:
Let's rephrase this to: "The first step in creating a Mixpanel source connection is to ensure that you have a Mixpanel account. If you do not have one already, see the [Mixpanel register](link) page to register and create your account."

carlojavier commented on help/sources/tutorials/ui/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,121 @@
+| Credential | Description | Example |
+| --- | --- | --- |
+| Host | Mixpanel Raw Data Export API endpoint. <br/><br/>Refer to the [Mixpanel API reference](https://developer.mixpanel.com/reference/overview) page, then scroll to the *Raw Data Export API* section if you require any guidance. | *https://data.mixpanel.com*|
+| Username | Service account Username. <br/><br/>Refer to the [Service Accounts](https://developer.mixpanel.com/reference/service-accounts#authenticating-with-a-service-account) page if you require any guidance. | *`xxxxxxxxxxx`.6d4ee7.mp-service-account* |
```
Any reason why only `xxxxxxxxxxx` is backticks? If it's means to hide sensitive values, we could maybe just randomize some characters instead of using `xxxxxxxxxxx`

carlojavier replied:
Any reason why only `xxxxxxxxxxx` is backticks? If it's means to hide sensitive values, we could maybe just randomize some characters instead of using `xxxxxxxxxxx`

carlojavier commented on help/sources/tutorials/ui/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,121 @@
+| --- | --- | --- |
+| Host | Mixpanel Raw Data Export API endpoint. <br/><br/>Refer to the [Mixpanel API reference](https://developer.mixpanel.com/reference/overview) page, then scroll to the *Raw Data Export API* section if you require any guidance. | *https://data.mixpanel.com*|
+| Username | Service account Username. <br/><br/>Refer to the [Service Accounts](https://developer.mixpanel.com/reference/service-accounts#authenticating-with-a-service-account) page if you require any guidance. | *`xxxxxxxxxxx`.6d4ee7.mp-service-account* |
+| Password | Service account Password. | *8iQbnjuk5.................* |
```
same here, maybe we could just randomize characters for the password since we're just showing format anyway

carlojavier replied:
same here, maybe we could just randomize characters for the password since we're just showing format anyway

carlojavier commented on help/sources/tutorials/ui/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,121 @@
+| Host | Mixpanel Raw Data Export API endpoint. <br/><br/>Refer to the [Mixpanel API reference](https://developer.mixpanel.com/reference/overview) page, then scroll to the *Raw Data Export API* section if you require any guidance. | *https://data.mixpanel.com*|
+| Username | Service account Username. <br/><br/>Refer to the [Service Accounts](https://developer.mixpanel.com/reference/service-accounts#authenticating-with-a-service-account) page if you require any guidance. | *`xxxxxxxxxxx`.6d4ee7.mp-service-account* |
+| Password | Service account Password. | *8iQbnjuk5.................* |
+| projectId | MixPanel Project Id. <br/><br/>Refer to the pages below if you require any guidance. <br/>[Project Settings](https://help.mixpanel.com/hc/en-us/articles/115004490503-Project-Settings) <br/>[Create and Manage Projects](https://help.mixpanel.com/hc/en-us/articles/115004505106-Create-and-Manage-Projects) | *2384945* |
```
Small typo, should be a lowercase "p" on Mixpanel

carlojavier replied:
Small typo, should be a lowercase "p" on Mixpanel

carlojavier commented on help/sources/tutorials/ui/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,121 @@
+| projectId | MixPanel Project Id. <br/><br/>Refer to the pages below if you require any guidance. <br/>[Project Settings](https://help.mixpanel.com/hc/en-us/articles/115004490503-Project-Settings) <br/>[Create and Manage Projects](https://help.mixpanel.com/hc/en-us/articles/115004505106-Create-and-Manage-Projects) | *2384945* |
+| timezone | Select the timezone to correspond to your project within the Mixpanel account. <br/><br/>Refer to the [Project Settings](https://help.mixpanel.com/hc/en-us/articles/115004490503-Project-Settings) page if you require any guidance.| *Pacific Standard Time* |
+
+The Mixpanel Project Id can be obtained by navigating to the screen below within your account.
```
Let's rephrase this to: "You can retrieve your Mixpanel Project ID and configure your project's timezone by navigating to the Project Details page of your account in the Mixpanel UI."

carlojavier replied:
Let's rephrase this to: "You can retrieve your Mixpanel Project ID and configure your Timezone by navigating to the Project Details page of your account in the Mixpanel UI."

carlojavier commented on help/sources/tutorials/ui/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,121 @@
+
+![Mixpanel Project Details](../../../../images/tutorials/create/mixpanel-export-events/mixpanel-project-settings.png?lang=en)
+
+The Mixpanel Service Account credentials can be obtained by navigating to the screen below within your account.
```
Similar rephrasing here: "To retrieve your Mixpanel service account credentials, navigate to the Service Accounts in the Mixpanel UI. 
Also a bit of a nitpick, but can we update this screenshot to one without the "Service Accounts" pop-up bubble appearing with the cursor? Thanks!

carlojavier replied:
Similar rephrasing here: "To retrieve your Mixpanel service account credentials, navigate to the Service Accounts in the Mixpanel UI. 
Also a bit of a nitpick, but can we update this screenshot to one without the "Service Accounts" pop-up bubble appearing with the cursor? Thanks!

carlojavier commented on help/sources/tutorials/ui/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,121 @@
+
+>[!TIP]
+>
+>* It is suggested to use a service account that doesn't expire.
```
Is this configurable by a user? Wondering if we need to add additional guidance to this

carlojavier replied:
Is this configurable by a user? Wondering if we need to add additional guidance to this

carlojavier commented on help/sources/tutorials/ui/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,121 @@
+
+You can select the appropriate category from the catalog on the left-hand side of your screen. Alternatively, you can find the specific source you wish to work with using the search option.
+
+Under the *Customer Success* category, select *Mixpanel*, and then select **[!UICONTROL Add data]**.
```
Update *Customer Success* to *Analytics*

carlojavier replied:
Update *Customer Success* to *Analytics*

carlojavier commented on help/sources/tutorials/ui/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,121 @@
+
+Next, add the projectId and select the timezone.
+
+![configuration](../../../../images/tutorials/create/mixpanel-export-events/authentication-configuration.png?lang=en)
```
Let's hide the IMS Organization on the top right of the screen

carlojavier replied:
Let's hide the IMS Organization on the top right of the screen

carlojavier commented on help/sources/tutorials/ui/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,121 @@
+   |data.mp_api_endpoint|string|
+   |data.mp_api_timestamp_ms|integer|
+   |data.mp_processing_time_ms|integer|
+   |data.time|integer|
```
Can we wrap all the source fields in backticks here?

carlojavier replied:
Can we wrap all the source fields in backticks here?

carlojavier reviewed (changes requested):
@mohan-h Thanks for this! review done for UI side of the docs. Will notify you once I finish API side.

carlojavier commented on help/sources/tutorials/api/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,1072 @@
+hide: true
+hidefromtoc: true
+---
+# Create a *Mixpanel* source connection in the UI
```
Title should be "Create a source connection and dataflow for Mixpanel using the Flow Service API"

mohan-h replied:
Oops sorry CPE

carlojavier commented on help/sources/tutorials/api/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,1072 @@
+---
+# Create a *Mixpanel* source connection in the UI
+
+This tutorial provides steps for creating a *Mixpanel* source connector using the Platform user interface.
```
Let's update this to: "The following tutorial walks you through the steps to create a source connection and a dataflow to bring Mixpanel data to Platform using the [Flow Service API](https://developer.adobe.com/experience-platform-apis/references/flow-service/).

mohan-h replied:
Oops sorry CPE.

carlojavier commented on help/sources/tutorials/api/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,1072 @@
+
+## Overview
+
+[Mixpanel](https://www.mixpanel.com) is a product analytics tool that enables capturing data on how users interact with a digital product. Mixpanel allows you to analyze this product data with simple, interactive reports that let you query and visualize the data with just a few clicks.
```
This high-level overview of the product is nice to have. Thanks for adding!

carlojavier commented on help/sources/tutorials/api/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,1072 @@
+
+## Prerequisites
+
+Before you start configuring the extension you need to have a Mixpanel account. If you do not have one already, go to the Mixpanel [register](https://mixpanel.com/register/) page to register and create your Mixpanel account.
```
Similar change with the UI-side here: "The first step in creating a Mixpanel source connection is to ensure that you have a Mixpanel account. If you do not have one already, see the [Mixpanel register](https://github.com/AdobeDocs/experience-platform.en/pull/64/link) page to register and create your account."

carlojavier commented on help/sources/tutorials/api/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,1072 @@
+
+[Mixpanel](https://www.mixpanel.com) is a product analytics tool that enables capturing data on how users interact with a digital product. Mixpanel allows you to analyze this product data with simple, interactive reports that let you query and visualize the data with just a few clicks.
+
+This Adobe Experience Platform [sources](https://experienceleague.adobe.com/docs/experience-platform/sources/home.html?lang=en) leverages the [Event Export API > Download](https://developer.mixpanel.com/reference/raw-event-export) to download your event data as it is received and stored within Mixpanel, along with all event properties (including distinct_id) and the exact timestamp the event was  ...
```
Same revision here: "The Mixpanel [source](https://github.com/AdobeDocs/experience-platform.en/pull/64/link) leverages..."

carlojavier commented on help/sources/tutorials/api/create/analytics/mixpanel.md:
````diff
@@ -0,0 +1,1072 @@
+
+```shell
+curl -X GET \
+    'https://platform.adobe.io/data/foundation/flowservice/connections/70383d02-2777-4be7-a309-9dd6eea1b46d/explore?objectType=rest&object=json&fileType=json&preview=true&sourceParams=eyJsaXN0SWQiOiIxMGMwOTdjYTcxIn0=' \
````
I think the values we have here for `list_id` and `source_params` are from Mailchimp Members. Can we update these to the Mixpanel ones? Values can just be example, but want to make sure the param for `list_id` is updated to the appropriate verbiage... unless it is also called `list_id` for Mixpanel?

carlojavier reviewed (changes requested):
@mohan-h Completed review of API-side. Just a few changes, thanks!

mohan-h commented:
Committed https://github.com/mohan-h/experience-platform.en/commit/eaae81af7fff3072e27e32d720d960b07177d57a addressing review inputs.

vivetiwa commented on help/sources/tutorials/api/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,1072 @@
+        },
+        "auth": {
+            "specName": "OAuth2 Refresh Code",
+            "params": {
```
JSON formatting could be added.

mohan-h replied:
Not sure about the json formatting as its a curl request and formatting is prefixed with the ```shell directive. Have added indentation.

vivetiwa commented on help/sources/tutorials/api/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,1072 @@
+| `{OBJECT}` | This parameter is required only when viewing a specific directory. Its value represents the path of the directory you wish to explore. |
+| `fileType=json` | The file type of the file you want to bring to Platform. Currently, `json` is the only supported file type. |
+| `{PREVIEW}` | A boolean value that defines whether the contents of the connection supports preview. |
+| `{SOURCE_PARAMS}` | Defines parameters for the source file you want to bring to Platform. To retrieve the accepted format-type for `{SOURCE_PARAMS}`, you must encode the entire `{"projectId":"2671127","timezone":"Pacific Standard Time"}` string in base64. In the example below, `"{"projectId":"2671127","timezone":"Pacific Standard Time"}": "10c097ca71"` encoded in base64 equates to `eyJwcm9qZWN0 ...
```
In line `{"projectId":"2671127","timezone":"Pacific Standard Time"}": "10c097ca71"` -`10c097ca71` is not required.

vivetiwa commented on help/sources/tutorials/api/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,1072 @@
+| --------- | ----------- |
+| `{BASE_CONNECTION_ID}` | The base connection ID generated in the previous step. |
+| `objectType=rest` | The type of object that you wish to explore. Currently, this value is always set to `rest`. |
+| `{OBJECT}` | This parameter is required only when viewing a specific directory. Its value represents the path of the directory you wish to explore. |
```
{OBJECT} could be json always in case of SSS sources.

vivetiwa commented on help/sources/tutorials/api/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,1072 @@
+        "description": "{Mixpanel} Source Connection",
+        "baseConnectionId": "70383d02-2777-4be7-a309-9dd6eea1b46d",
+        "connectionSpec": {
+            "id": "6360f136-5980-4111-8bdf-15d29eab3b5a",
```
Final connectionSpec id would be `fd2c8ff3-1de0-4f6b-8fa8-4264784870eb`

mohan-h replied:
@vivetiwa There are 4 occurrences of connectionSpec. Changed for all 4 occurrences. 
However within [zendesk api](https://experienceleague.adobe.com/docs/experience-platform/sources/api-tutorials/create/customer-success/zendesk.html?lang=en) only 2 id's match-up. Could you let me know which instances to change.

cc @carlojavier

vivetiwa commented on help/sources/tutorials/api/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,1072 @@
+        },
+        "params": {
+            "projectId": "{MIXPANEL_PROJECT_ID}",
+            "timezone": "{MIXPANEL_PROJECT_ID}"
```
typo MIXPANEL_PROJECT_ID should be MIXPANEL_TIMEZONE

mohan-h replied:
Oops, good spot.

vivetiwa commented on help/sources/tutorials/api/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,1072 @@
+| `description` | An optional value that you can include to provide more information on your source connection. |
+| `baseConnectionId` | The base connection ID of *Mixpanel*. This ID was generated in an earlier step. |
+| `connectionSpec.id` | The connection specification ID that corresponds to your source. |
+| `data.format` | The format of the *Mixpanel* data that you want to ingest. Currently, the only supported data format is `json`. |
```
projectId and timezone parameter missing in table

vivetiwa commented on help/sources/tutorials/ui/create/analytics/mixpanel.md:
```diff
@@ -0,0 +1,121 @@
+
+![existing](../../../../images/tutorials/create/mixpanel-export-events/authentication-existing-account.png?lang=en)
+
+### New account
```
After new account can we add preview screen with input value.

juliette-massardier committed:
Update license-usage.md

Hello, as discussed with the Adobe Journey Optimizer product team, we should add the AJO licenses listed here and the link to the AJO product description. Thanks! Juliette (AJO doc team)

JCON3DEV committed:
Merge pull request #80 from juliette-massardier/patch-1

Update AJO license-usage.md

mohan-h committed:
Merge branch 'AdobeDocs:main' into mixpanel-source

mohan-h commented:
The documentation for the connector has been made live at:
https://experienceleague.adobe.com/docs/experience-platform/sources/connectors/analytics/mixpanel.html?lang=en
